### DISCUSSION TITLE
marimo integration with Safe Exam Browser

### DISCUSSION THREAD

**Discussion by VivaldoMendes** [marimo-team/marimo#3025] [Q&A]
### Description

Hi. For conducting teaching and assessments using marimo, it is becoming critical to have some configuration that would preclude students from cheating during assessment moments.  Launching the Safe Exam Browser software (SEB) in the Moodle platform is a well-known process for implementing that security safeguard.  However, the Moodle platform has severe limitations if we want to use a notebook like marimo. Is it possible to create a configuration that would preclude students from opening a new window when marimo is running during a test/exam? For teaching/grading, this functionality would be a game changer.

### Suggested solution

Create a configuration that would preclude students from opening a new window when marimo is running during a test/exam.

**Comment by mscolnick** [comment]
marimo is run in a browser and does not take control of the browser (it does not have any more special permissions than any other site), so unfortunately, I am not sure this would be possible in marimo alone.

**Comment by VivaldoMendes** [comment]
I understand.  This problem will become a big challenge for those who use notebooks (jupyter, marimo, pluto, and the like) in their teaching and assessment moments when many students are involved. Teaching using a notebook and taking the tests on paper does not make sense.  Thank you very much for clarifying this issue.

**Comment by mscolnick** [comment]
@VivaldoMendes Did you use Jupyter before marimo? What did you do before for mimicking "safe exam browser"?

**Comment by VivaldoMendes** [comment]
@mscolnick  I stoped using Jupyter in 2020, on a regular basis because of no-reacticity. I have been using Pluto quite a lot since then, and since last July I have also been using marimo. Currently, I have two courses running on Pluto notebooks, and I tried to implement some form of integration between Pluto and Safe Exam Browser, but I failed.

**Comment by mscolnick** [comment]
Thanks for the explanation. I am going to move this to a discussion since I don't think there is any action/solution on our end at the moment.
